src/general/DayNightCycle.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using Newtonsoft.Json;
+
+[JsonObject(IsReference = true)]
+[UseThriveSerializer]
+public class DayNightCycle
+{
+    [JsonProperty]
+    public DayNightConfiguration LightCycleConfig;
+
+    [JsonProperty]
+    public bool IsEnabled;
+
+    /// <summary>
+    ///   The multiplier used for calculating DayLightPercentage.
+    /// </summary>
+    /// <remarks>
+    ///   This exists as it only needs to be calculated once and
+    ///   the calculation for it is confusing.
+    /// </remarks>
+    [JsonIgnore]
+    private float daytimeMultiplier;
+
+    public DayNightCycle(bool isEnabled)
+    {
+        IsEnabled = isEnabled;
+
+        LightCycleConfig = SimulationParameters.Instance.GetDayNightCycleConfiguration();
+        Time = LightCycleConfig.HoursPerDay / 2;
+
+        float halfPercentage = LightCycleConfig.DaytimePercentage / 2;
```

Comment:
This `halfPercentage` name does not tell at all that were are dealing with the (scaled) duration of daytime. 

As a note, I also was confused by the name `DaytimePercentage` as well. I assumed it would represent the max value of sunlight at day time (which is a percentage). Perhaps the term "percentage" isn't great as it refers to a measure unit and not a "dimension"?

Reply:
DaytimePercentage is the Percentage of the day that has daylight.

Reply:
To me DaytimePercentage would be how much current light is of the daytime value. I'd use a name like `DaytimeDurationPercentage` or `DaylightDurationPercentage` or something like that.

Reply:
I second this suggestion. I finally understood what DaytimePercentage meant, but as I stated above the name is indeed misleading.

`halfPercentage` could also still use a clearer name even though you only use it locally.